### PR TITLE
chore(Toolbar): convert ToolbarDivider, ToolbarCustomItem & ToolbarItem to functional components

### PR DESCRIPTION
**Pull request by layershifter**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ yarn change`

# BREAKING CHANGES

This PR converts `ToolbarDivider`, `ToolbarCustomItem` & `ToolbarItem` components to be functional. As in all other cases it contains a breaking change with restricted props set that will be passed to styles functions.

## Prop sets

| `ToolbarDivider`    |
| --------- |
| no props |

| `ToolbarCustomItem`      |
| --------------- |
| `fitted`          |

| `ToolbarItem`      |
| --------------- |
| `active`          |
| `disabled`          |

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/12161)

### CODE REVIEW

File: packages/fluentui/accessibility/src/behaviors/Toolbar/toolbarItemBehavior.ts [hunk 1]
```diff
@@ -39,9 +40,9 @@ export default toolbarItemBehavior;
 
 export type ToolbarItemBehaviorProps = {
   /** Indicated if toolbar item has a menu. */
-  menu?: boolean | object;
+  hasMenu?: boolean;
```
Comment (mnajdova) [inline packages/fluentui/accessibility/src/behaviors/Toolbar/toolbarItemBehavior.ts]: 👍

File: packages/fluentui/react/src/components/Toolbar/ToolbarDivider.tsx [hunk 2]
```diff
@@ -1,38 +1,52 @@
...
+  });
 
-  renderComponent({ ElementType, classes, unhandledProps, accessibility }) {
-    return <ElementType {...accessibility.attributes.root} {...unhandledProps} className={classes.root} />;
-  }
-}
+  const ElementType = getElementType(props);
+  const unhandledProps = getUnhandledProps(ToolbarDivider.handledProps, props);
+
+  const element = <ElementType {...getA11yProps('root', { className: classes.root, ...unhandledProps })} />;
```
Comment (mnajdova) [inline packages/fluentui/react/src/components/Toolbar/ToolbarDivider.tsx]:
```suggestion
  const element = <ElementType {...getA11yProps('root', { ...unhandledProps, className: classes.root, })} />;
```

File: packages/fluentui/react/src/themes/teams/types.ts [hunk 3]
```diff
@@ -105,8 +106,9 @@ export type TeamsThemeStylesProps = {
   Slider?: SliderStylesProps;
   Status?: StatusStylesProps;
   Toolbar?: ToolbarProps;
-  ToolbarItem?: ToolbarItemProps;
-  ToolbarDivider?: ToolbarDividerProps;
+  ToolbarCustomItem?: ToolbarCustomItemProps;
```
Comment (mnajdova) [inline packages/fluentui/react/src/themes/teams/types.ts]:
```suggestion
  ToolbarCustomItem?: ToolbarCustomItemStylesProps;
```

File: packages/fluentui/react/src/themes/teams/types.ts [hunk 4]
```diff
@@ -44,8 +44,9 @@ import { SegmentProps } from '../../components/Segment/Segment';
 import { SliderStylesProps } from './components/Slider/sliderStyles';
 import { StatusStylesProps } from './components/Status/statusStyles';
 import { TextStylesProps } from './components/Text/textStyles';
-import { ToolbarDividerProps } from '../../components/Toolbar/ToolbarDivider';
-import { ToolbarItemProps } from '../../components/Toolbar/ToolbarItem';
+import { ToolbarDividerStylesProps } from '../../components/Toolbar/ToolbarDivider';
+import { ToolbarItemStylesProps } from '../../components/Toolbar/ToolbarItem';
+import { ToolbarCustomItemProps } from '../../components/Toolbar/ToolbarCustomItem';
```
Comment (mnajdova) [inline packages/fluentui/react/src/themes/teams/types.ts]:
```suggestion
import { ToolbarCustomItemStylesProps } from '../../components/Toolbar/ToolbarCustomItem';
```

File: packages/fluentui/react/src/components/Toolbar/ToolbarCustomItem.tsx [hunk 5]
```diff
@@ -46,49 +46,70 @@ export interface ToolbarCustomItemProps extends UIComponentProps, ChildrenCompon
   onBlur?: ComponentEventHandler<ToolbarCustomItemProps>;
 }
 
-class ToolbarCustomItem extends UIComponent<WithAsProp<ToolbarCustomItemProps>> {
-  static displayName = 'ToolbarCustomItem';
+export type ToolbarCustomItemStylesProps = Required<Pick<ToolbarCustomItemProps, 'fitted'>>;
```
Comment (pompomon) [inline packages/fluentui/react/src/components/Toolbar/ToolbarCustomItem.tsx]: Please, why do you set 'fitted' as a required style prop?

**Pull request merged**